pkg/kubelet/cri/remote/remote_runtime.go

```diff
@@ -796,5 +797,25 @@ func (r *remoteRuntimeService) CheckpointContainer(options *runtimeapi.Checkpoin
 }
 
 func (r *remoteRuntimeService) GetContainerEvents(containerEventsCh chan *runtimeapi.ContainerEventResponse) error {
-	return nil
+	containerEventsStreamingClient, err := r.runtimeClient.GetContainerEvents(context.Background(), &runtimeapi.GetEventsRequest{})
+
+	if err != nil {
+		klog.ErrorS(err, "GetContainerEvents failed to get streaming client")
+		return err
+	}
+
+	for {
+		resp, err := containerEventsStreamingClient.Recv()
+		if err == io.EOF {
+			return err
+		}
+		if err != nil {
+			klog.ErrorS(err, "cannot receive container event")
+			return err
+		}
+		if resp != nil {
+			containerEventsCh <- resp
```

Comment:
The resp != nil can be removed since resp cannot ever be nil.

Reply:
Yeah, the `resp` cannot ever be nil in this case. However, considering the return value of the `containerEventsStreamingClient.Recv()` which is a pointer it could be a good defensive practice to still check for `nil` here. That way, we won't be dependent on the implementation of `Recv()` to handle the nil pointer for us.